integration-tests/tests/di_diun.rs

```diff
@@ -71,3 +75,91 @@ async fn test_device_credentials_already_active() -> Result<()> {
 
     Ok(())
 }
+
+#[tokio::test]
+async fn test_device_credentials_generated_with_mac_address() -> Result<()> {
```

Comment:
Here can we add a test inside by running manufacturing client separately as `plain DI` and `DIUN` , so that we cover both cases